src/components/AccountCard.vue

```diff
@@ -152,28 +156,56 @@ export default defineComponent({
             }
         };
 
-        const loadResources = () => {
-            ram_used.value = fixDec(
-                Number(accountData.value.ram_usage) / KILO_UNIT.value,
-            );
+        const determineRamUnit = (size: number) => {
+            if (size > GIGA_UNIT.value) {
+                return {
+                    unit: 'gb',
+                    denominator: GIGA_UNIT.value,
+                };
+            }
+            if (size > MEGA_UNIT.value) {
+                return {
+                    unit: 'mb',
+                    denominator: MEGA_UNIT.value,
+                };
+            }
+            return {
+                unit: 'kb',
+                denominator: KILO_UNIT.value,
+            };
+        };
 
+        const loadResources = () => {
+            let ramDenominator;
             if (props.account !== system_account.value) {
+                // in most cases the max value unit is more im portant than the
+                // used value unit. So we use the max value unit.
+                // get max ram number
+                const ramMaxNumber = Number(accountData.value.ram_quota);
+                // get denominator based on max number
+                ramDenominator = determineRamUnit(ramMaxNumber).denominator;
+                // get displayed number value
                 ram_max.value = fixDec(
-                    Number(accountData.value.ram_quota) / KILO_UNIT.value,
+                    ramMaxNumber / ramDenominator,
+                );
+                // get units
+                ramUnit.value = determineRamUnit(ramMaxNumber).unit;
+                const netUsedNumber = Number(accountData.value.net_limit.used);
+                const netMaxNumber = Number(accountData.value.net_limit.max);
+                const netDenominator = determineRamUnit(netMaxNumber).denominator;
```

Comment:
```suggestion
                const netUnitResult = determineUnit(netMaxNumber);
                const netDenominator = netUnitResult.denominator;
                netUnit.value = netUnitResult.unit;
```
You only need to call determineUnit once with the same input as the input value never changes.